src/levanter/eval.py

```diff
@@ -479,3 +482,49 @@ def zeros_like(total: Arrayish, per_tag: Arrayish) -> "_EvalRunningMeans":
         z = RunningMean.zeros_like(total)
         per_tag = RunningMean.zeros_like(per_tag)
         return _EvalRunningMeans(z, per_tag, z, per_tag)
+
+
+@dataclass
+class EvalPluginConfig:
+    """Configuration for evaluation plugins."""
+
+    plugin_class: str
+    """Full module path to the plugin class (e.g., 'my.module.MyPlugin')."""
+    steps: Optional[int] = None
+    """Number of steps between plugin evaluations. If None, uses trainer's steps_per_eval."""
+
+
+class EvalPlugin(ABC):
+    """Abstract base class for evaluation plugins."""
+
+    @abstractmethod
+    def create_callback(
+        self,
+        *,
+        tokenizer: HfTokenizer,
```

Comment:
Passing through the tokenizer and batch size made sense to me here.  I'm not so sure about the device_mesh and axis mappings -- presumably at least the device mesh should be used for doing inference in parallel batches?

Overall, I don't really know what context is best -- guidance is welcome.

Reply:
yeah i think you want all these things. The Mesh is usually available via a global function at this point but everything else is pretty

What do you think the best way to deal with this kind of thing is? For, e.g., our audio models (which exist now!) we wouldn't want to just pass in a tokenizer, but a Processor... 

I've always struggled with how to expose this kind of stuff